__tests__/app.test.js

```diff
@@ -29,61 +29,93 @@ describe("GET /api", () => {
+          expect(typeof article.title).toEqual("string");
+          expect(typeof article.body).toEqual("string");
+          expect(typeof article.body).toEqual("string");
+          expect(typeof article.author).toEqual("string");
+          expect(typeof article.created_at).toEqual("string");
+          expect(typeof article.votes).toEqual("number");
+          expect(typeof article.article_img_url).toEqual("string");
+        });
+      });
+  });
+  test("404: Responds with an error message when the correct data type is requested but is currently non existent", () => {
+    return request(app)
+      .get("/api/article/999")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Article not found");
+      });
+  });
+  test("400: Responds with a bad request when a non suitable input request is made i.e not a number", () => {
+    return request(app)
+      .get("/api/article/abde")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("bad request, not a valid input");
       });
   });
 });
-test("404: Responds with an error message when the correct data type is requested but is currently non existent", () => {
-  return request(app)
-    .get('/api/article/999')
-    .expect(404)
-    .then((response) => {
-      expect(response.body.msg).toBe("Article not found");
-    });
-});
-test("400: Responds with a bad request when a non suitable input request is made i.e not a number", () => {
-  return request(app)
-    .get('/api/article/abde')
-    .expect(400)
-    .then((response) => {
-      expect(response.body.msg).toBe("bad request, not a valid input");
-    });
+describe("GET /api/articles", () => {
+  test("200:responds with an articles array of article objects ", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const body = response.text;
+        const articelArray = JSON.parse(body)
+        expect(Array.isArray(articelArray)).toBe(true);
+        articelArray.forEach((article) => {
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+          expect(typeof article.comment_count).toBe("number");
+        });
+      });
+  });
+  test("200:check if array is ordered in DESC order - convert date string to numbers ", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const body = response.text;
+        const articelArray = JSON.parse(body)
+        expect(
+          articelArray.map((a) => (a.created_at = new Date(a.created_at).getTime()))
```

Comment:
jest sorted should be able to check the original created_at for us, without us having to convert it!